File: 6_kyu/Are_they_the_same.go
package kata
import "sort"

func Comp(array1 []int, array2 []int) bool {
    if array1 == nil || array2 == nil { return false }
    if len(array1) != len(array2) { return false }
    for i,v := range array1 {
        if v < 0 { array1[i] = -v }
    }
    sort.Ints(array1)
    sort.Ints(array2)
    for i,v := range array1 {
        if v*v != array2[i] { return false }
    }
    return true
}

File: 6_kyu/Are_they_the_same.py
def comp(array1 :list[int], array2 :list[int]) -> bool:
    """
    Returns true if for each item in array1 an item² in array2 exists.
    """
    if not array1 or not array2:
        return array1 == array2
    return sorted(x**2 for x in array1) == sorted(array2)
